superset/sql_parse.py

```diff
@@ -77,6 +77,29 @@ def _extract_limit_from_query(statement: TokenList) -> Optional[int]:
                 return int(token.value)
     return None
 
+def _extract_top_from_query(statement: TokenList) -> Optional[int]:
+    """
+    Extract top clause value from SQL statement.
+
+    :param statement: SQL statement
+    :return: top value extracted from query, None if no top value present in statement
+    """
+
+    td_top_keyword = {"TOP","SAMPLE"}
+    str_statement = str(statement)
+    str_statement = str_statement.replace("\n", " ").replace("\r", "")
+    token = str_statement.rstrip().split(" ")
+    token = [part for part in token if part]
+    top = None
+    for i, _ in enumerate(token):
+        if token[i].upper() in td_top_keyword and len(token) - 1 > i:
+            try:
+                top = int(token[i + 1])
+            except ValueError:
+                top = None
+            break
+    return top
```

Comment:
Here we'd need to pass in the `top_keywords` from the db engine spec to make it possible to customize supported keywords per engine. Something like this:
```suggestion
def _extract_top_from_query(statement: TokenList, top_keywords: Set[str]) -> Optional[int]:
    """
    Extract top clause value from SQL statement.

    :param statement: SQL statement
    :return: top value extracted from query, None if no top value present in statement
    """

    str_statement = str(statement)
    str_statement = str_statement.replace("\n", " ").replace("\r", "")
    token = str_statement.rstrip().split(" ")
    token = [part for part in token if part]
    top = None
    for i, _ in enumerate(token):
        if token[i].upper() in top_keywords and len(token) - 1 > i:
            try:
                top = int(token[i + 1])
            except ValueError:
                top = None
            break
    return top
```